internal/tuplemapper/tuplemapper.go

```diff
@@ -0,0 +1,56 @@
+package tuplemapper
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/tuple"
+)
+
+type MapperKind int64
+
+const (
+	// NestedUsersetKind is a mapper that returns the userset ID from the tuple's user field.
```

Comment:
comment (non-blocking): NestedUsersetKind / NestedTTUKind does not feel like belonging here.  It is exclusively used in graph/check.go

Reply:
moved